tests/functionality_tests/sensor_physical_testing.py

```diff
@@ -0,0 +1,42 @@
+from engine.robot_state import Robot_State
+from engine.robot import Robot
+from electrical.gps import GPS
+from engine.grid import Grid
+from engine. database import DataBase
+import math
+
+if __name__ == "__main__":
+    # make sure to go to imu and gps and set import to true
+    test_state = 0  # 0 is no node straight, 1 is node straight, 2 is turn, -1 is sensor
```

Comment:
what is the difference between no node straight and node straight?

Reply:
no node straight is PID without using node from grid. node straight is generating the node from the grid and using that for PID

Reply:
so there aren't any waypoints in no node straight?

Reply:
nope, unless you count the `[10, 10]` in `rb.move_to_target_node([10, 10], 2, database)`